### PR TITLE
Use Github Actions for CI

### PR DESCRIPTION
**Pull request by pmalek-sumo**
###### Description

Start using Github Actions for CI.
For now it works for PRs but further work is required to handle github environment variables for pushing docker images purposes.

###### Testing performed

- [ ] ci/build.sh
- [ ] Redeploy fluentd and fluentd-events pods
- [ ] Confirm events, logs, and metrics are coming in

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,2 @@
+shellcheck:
+	./ci/shellcheck.sh
```
Comment (sumo-drosiek) [inline Makefile:2]: add to vagrant as well

Comment (sumo-drosiek) [reply Makefile:2]: also add it to `test` target or create new one to perform all required tests

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        with:
+          ruby-version: '2.6'
+      - run: gem install bundler
+      - name: yq
+        run: |
+            curl --retry 10 --retry-max-time 120 --retry-delay 5 -L -o /tmp/yq https://github.com/mikefarah/yq/releases/download/3.4.1/yq_linux_amd64
+            chmod +x /tmp/yq
+            sudo mv /tmp/yq /usr/local/bin/yq
+      - name: tests
+        run: bash ci/tests.sh
```
Comment (sumo-drosiek) [inline .github/workflows/main.yml:51]: we could use

```suggestion
        run: make test
```

maybe 🤔

File: ci/build.sh [hunk 3]
```diff
@@ -72,8 +72,8 @@ function set_up_github() {
   local token="${1}"
   local branch="${2}"
 
-  git config --global user.email "travis@travis-ci.org"
-  git config --global user.name "Travis CI"
+  git config --global user.email "bot@github.com"
+  git config --global user.name "Github Action"
```
Comment (sumo-drosiek) [inline ci/build.sh:76]: shouldn't we use platform independent values?
I would skip email and set `user.name` to `Continuous Integration`

Comment (pmalek-sumo) [reply ci/build.sh]: Done, it doesn't work now anyway because of not handling env variables that contain branch name etc. in GH env

**Review by sumo-drosiek — APPROVED** [review]

**Review by perk-sumo — APPROVED** [review]
👍

**Pull request merged**